app/src/test/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java

```diff
@@ -199,21 +199,34 @@ public void deletePost(User user, Forum forum, DateTime postCreationDate) throws
             throw new ForumNotFoundException();
         }
         for (Group g : groups) {
-            if (g.getName().equals(forum.getGroup().getName())) {
-                for (Forum f : g.getForums()) {
-                    if (f.getName().equals(forum.getName())) {
-                        boolean found = false;
-                        for (Post p : f.getPosts()) {
-                            if (p.getUsername().equals(user.getUsername())
-                                && p.getCreationDate().compareTo(postCreationDate) == 0) {
-                                found = true;
-                                f.removePost(user, postCreationDate);
-                            }
-                        }
-                        if (!found) {
-                            throw new PostNotFoundException();
+            deleteGroupPosts(user, forum, postCreationDate, g);
+        }
+    }
+
+    /**
+     * Deletes the posts form a group.
+     * @param user The user
+     * @param forum The forum
+     * @param postCreationDate The creation date of a post
+     * @param group The group
+     * @throws PostNotFoundException The post is not found
+     */
+    private void deleteGroupPosts(User user, Forum forum, DateTime postCreationDate, Group group)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method length is 19 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5421171)